mysql-test/main/mysql_tzinfo_to_sql_symlink.test

```diff
@@ -3,7 +3,8 @@
 --source include/not_windows.inc
 --source include/no_protocol.inc
 
-let $is_embedded=`select version() like '%embedded%'`;
+# In certain build configurations (e.g. embedded server), these variables will have the value NULL.
+let $replace_wsrep_cols=`select @wsrep_is_on is NULL or @wsrep_cannot_replicate_tz`;
```

Comment:
Ultimately, the fact that these variables can be null comes from the way that they are defined, as of https://github.com/MariaDB/server/commit/13e77930e615f05cc74d408110e887b00e1abcc9: 

https://github.com/MariaDB/server/blob/13e77930e615f05cc74d408110e887b00e1abcc9/sql/tztime.cc#L2738-L2743

https://github.com/MariaDB/server/blob/13e77930e615f05cc74d408110e887b00e1abcc9/sql/tztime.cc#L2767-L2781

If Galera (aka "wsrep"?) is not compiled in, this variable doesn't exist _at all_, leading to `@wsrep_is_on=NULL` → `@wsrep_cannot_replicate_tz=NULL`.

So the cleanest and most robust fix would be to change those initial definitions to include `coalesce(…, 0)`, and then get rid of the special cases in the test.

@grooverdan, do you agree with that? Is there any reason that any consumer of the tzinfo SQL would actually need to differentiate between "Galera is OFF" and "Galera is not compiled in"?

```c
  printf("set @wsrep_is_on=(%s);\n", wsrep_is_on);
```

Reply:
`NULL` is also a `false` behaviour in the sql standard, so I saw no need for `coalesce`, but you're right in that didn't consider this in the test flexibility.

Please add `select version() like '%embedded%'` into a `or` condition of `$replace_wsrep_cols`. I don't think you intentionally removed this?

Reply:
> Please add `select version() like '%embedded%'` into a `or` condition of `$replace_wsrep_cols`. I don't think you intentionally removed this?

I _did_ mean to remove that, actually. It seems to me to be error-prone and indirect to try to infer whether or not Galera is supported by looking at the version string, rather than simply looking at whether or not the Galera-related variables are populated (as I'm proposing here).

Reply:
@grooverdan, in 0e4cf497ca11a7298e2bd896cb594bd52085a1d4 you modified the `SELECT` query for `wsrep_is_on` to work around issues in bootstrap mode:

```diff
--- a/sql/tztime.cc
+++ b/sql/tztime.cc
@@ -2726,11 +2726,11 @@ static const char *trunc_tables_const=
   "TRUNCATE TABLE time_zone_transition;\n"
   "TRUNCATE TABLE time_zone_transition_type;\n";
 static const char *wsrep_is_on=
-  "select sum(VARIABLE_NAME='wsrep_on' AND SESSION_VALUE='ON')"
-  " from information_schema.SYSTEM_VARIABLES";
+  "select sum(SESSION_VALUE='ON')"
+  " from information_schema.SYSTEM_VARIABLES WHERE VARIABLE_NAME='wsrep_on'";
 static const char *wsrep_cannot_replicate_tz=
-  "select sum(VARIABLE_NAME='wsrep_mode' and GLOBAL_VALUE NOT LIKE @replicate_opt)"
-  " from information_schema.SYSTEM_VARIABLES";
+  "select sum(GLOBAL_VALUE NOT LIKE @replicate_opt)"
+  " from information_schema.SYSTEM_VARIABLES WHERE VARIABLE_NAME='wsrep_mode'";
 
 int
 main(int argc, char **argv)
```